cpp/src/groupby/new_groupby.cu

```diff
@@ -269,17 +271,28 @@ gdf_unique_indices(cudf::table const& input_table, gdf_context const& context)
   return unique_indices;
 }
 
-std::pair<cudf::table, rmm::device_vector<gdf_index_type>> 
+std::pair<cudf::table, gdf_column>
 gdf_group_by_without_aggregations(cudf::table const& input_table,
                                   gdf_size_type num_key_cols,
                                   gdf_index_type const * key_col_indices,
                                   gdf_context* context)
 {
   CUDF_EXPECTS(nullptr != key_col_indices, "key_col_indices is null");
   CUDF_EXPECTS(0 < num_key_cols, "number of key colums should be greater than zero");
+  gdf_column unique_indices;
```

Comment:
You don't need to manually zero out all of the members if you use this form of construction. 
```suggestion
  gdf_column unique_indices{};
```